Wrong C code generated for inline specified fixed array

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
V full version: V 0.4.4 ac2dcc2
OS: linux, "EndeavourOS Linux"
Processor: 12 cpus, 64bit, little endian, AMD Ryzen 5 PRO 4650U with Radeon Graphics

getwd: /home/odi/Temp
vexe: /home/odi/Repos/vlang/v
vexe mtime: 2024-01-09 14:37:16

vroot: OK, value: /home/odi/Repos/vlang
VMODULES: OK, value: /home/odi/.vmodules
VTMP: OK, value: /tmp/v_1000

Git version: git version 2.43.0
Git vroot status: 0.4.4
.git/config present: true

CC version: cc (GCC) 13.2.1 20230801
thirdparty/tcc status: thirdparty-linux-amd64 12f392c3

```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg bug.v`
```v
struct Foo {
    bar &&char
}

fn main() {
    foo := Foo{
        bar: unsafe {
            &&char(['a', 'b', nil]!)
        }
    }
    println(foo)
}

```

**What did you expect to see?**

Equivalent of char** in C correctly defined in V.

**What did you see instead?**
```
bug.v:8:13: warning: cannot cast a fixed array (use e.g. `&arr[0]` instead)
    6 |     foo := Foo{
    7 |         bar: unsafe {
    8 |             &&char(['a', 'b', nil]!)
      |             ~~~~~~~~~~~~~~~~~~~~~~~~
    9 |         }
   10 |     }
==================
/tmp/v_1000/bug.16169138198730202255.tmp.c:12700: error: ';' expected (got "}")
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

Discussed before here: https://discord.com/channels/592103645835821068/592294828432424960/1193173809767927868

Generated C code that causes problems
```c
VV_LOCAL_SYMBOL void main__main(void) {
	main__Foo foo = ((main__Foo){.bar = ((char**)({_SLIT("a"), _SLIT("b"), ((void*)0)})),});
        //                                                                               ^
	println(main__Foo_str(foo));
}
```

> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/gen/c/array.v
+++ b/vlib/v/gen/c/array.v
@@ -146,6 +146,10 @@ fn (mut g Gen) fixed_array_init(node ast.ArrayInit, array_type Type, var_name st
 		ret_typ := g.typ(node.typ)
 		g.write('${ret_typ} ${tmp_var} = ')
 	}
+	if g.inside_cast {
+		ret_typ := g.typ(node.typ)
+		g.write('(${ret_typ})')
+	}
 	g.write('{')
 	if node.has_val {
 		tmp_inside_array := g.inside_array_item

--- a/vlib/v/gen/c/array.v
+++ b/vlib/v/gen/c/array.v
@@ -146,7 +146,7 @@ fn (mut g Gen) fixed_array_init(node ast.ArrayInit, array_type Type, var_name st
 		ret_typ := g.typ(node.typ)
 		g.write('${ret_typ} ${tmp_var} = ')
 	}
-	if g.inside_cast {
+	if g.inside_struct_init && g.inside_cast {
 		ret_typ := g.typ(node.typ)
 		g.write('(${ret_typ})')
 	}

--- a/vlib/v/tests/cast_fixed_array_to_ptr_ptr_test.v
+++ b/vlib/v/tests/cast_fixed_array_to_ptr_ptr_test.v
@@ -1,0 +1,14 @@
+struct Foo {
+	bar &&char
+}
+
+fn test_main() {
+	foo := Foo{
+		bar: unsafe {
+			&&char(['a', 'b', nil]!)
+		}
+	}
+	println(foo)
+	a := unsafe { cstring_to_vstring(*foo.bar) }
+	assert a == 'a'
+}

